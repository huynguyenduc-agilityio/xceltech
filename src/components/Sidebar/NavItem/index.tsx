import { ComponentType, SVGProps } from 'react';

// Utils
import { cn } from '@/utils';

type TNavItemProps = {
  subTitle: string;
  url: string;
  icon: ComponentType<SVGProps<SVGSVGElement>>;
};

const NavItem = ({ subTitle, url, icon: Icon }: TNavItemProps) => {
  const isDefaultFocused = location.pathname === '/';
  const isFocused = location.pathname === url;

  const linkClasses = cn(
    'flex items-center w-full gap-3 px-5 py-4 rounded-[14px] rounded-lg transition-colors',
    url ? 'cursor-pointer' : 'cursor-not-allowed',
    (isDefaultFocused || isFocused) && url
      ? 'bg-secondary text-black-default'
      : 'hover:bg-gray-ash',
  );
  return (
    <div className={linkClasses}>
      <div className="w-10">
        <Icon
          className={cn(
            (isDefaultFocused || isFocused) && url
              ? 'text-black'
              : 'text-white',
          )}
        />
      </div>
      <span className="text-md">{subTitle}</span>
    </div>
  );
};

export default NavItem;
