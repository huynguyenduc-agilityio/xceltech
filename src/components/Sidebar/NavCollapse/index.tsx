// Constants
import { ADMIN_NAVIGATION_CONFIG } from '@/constants';

import NavItem from '../NavItem';

const NavCollapse = () => {
  return (
    <div className="grid gap-2 px-4">
      {ADMIN_NAVIGATION_CONFIG.map(({ title, navItem }) => (
        <div key={title}>
          <div className="flex flex-start ml-5 h-[30px]">
            <span className="text-base">{title}</span>
          </div>
          <div className="grid gap-[10px]">
            {navItem.map(({ title, icon, url }) => (
              <NavItem key={title} subTitle={title} url={url} icon={icon} />
            ))}
          </div>
        </div>
      ))}
    </div>
  );
};

export default NavCollapse;
