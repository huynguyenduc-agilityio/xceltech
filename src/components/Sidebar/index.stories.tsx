import type { Meta, StoryObj } from '@storybook/react';

// Components
import Sidebar from '.';

const meta = {
  title: 'Components/Sidebar',
  component: Sidebar,
  parameters: {
    layout: 'centered',
  },

  tags: ['autodocs'],
} satisfies Meta<typeof Sidebar>;

export default meta;

type Story = StoryObj<typeof Sidebar>;

export const Default: Story = {
  args: {
    children: (
      <p>
        Lorem ipsum dolor, sit amet consectetur adipisicing elit. Delectus
        pariatur consectetur veritatis voluptatum culpa accusamus ipsam
        doloremque eveniet, natus ea ratione. Facere, sapiente eaque! Fugiat
        quam beatae temporibus vitae incidunt!
      </p>
    ),
  },
};
