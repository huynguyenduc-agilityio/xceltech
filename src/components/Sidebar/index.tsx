import { ReactNode } from 'react';
import { useNavigate } from 'react-router-dom';

// Constants
import { AUTHENTICATION_PAGE, IMAGES } from '@/constants';

// Stores
import { useUserActions } from '@/stores';

// Hooks
import { useGetInfoUser } from '@/hooks';

// Utils
import { getInitialsAvatar } from '@/utils';

// Icons
import { LogoIcon, LogoutIcon } from '@/icons';

// Component
import NavCollapse from './NavCollapse';
import { Avatar } from '../common';

const Sidebar = ({ children }: { children: ReactNode }) => {
  const navigate = useNavigate();
  const { userInfo } = useGetInfoUser();
  const { clearUser } = useUserActions();

  const { firstName = '', lastName = '', avatar, role } = userInfo || {};

  const handleClickLogoutBtn = () => {
    clearUser();
    navigate(AUTHENTICATION_PAGE.ADMIN_SIGN_IN);
  };

  return (
    <div className="flex min-h-screen w-full">
      {/* Navbar */}
      <nav className="flex flex-col justify-between fixed h-screen w-[359px] pt-[18px] overflow-y-auto transition-all duration-300 bg-blue-deepest text-white">
        <div>
          <LogoIcon />

          <div className="flex items-center mt-[47px] mb-[44px] px-4">
            <div className="w-[112px] h-[112px] p-2 rounded-full border-2 border-white">
              <Avatar
                src={avatar as string}
                fallback={getInitialsAvatar(`${firstName} ${lastName}`)}
                size="100%"
              />
            </div>
            <div className="flex flex-col ml-[22px]">
              <span className="text-xl font-bold">
                {firstName} {lastName}
              </span>
              <span className="text-sm">{role}</span>
            </div>
          </div>

          <NavCollapse />
          <div className="mt-12 mb-3 px-4">
            <button
              className="flex items-center justify-center py-4 bg-red-600 rounded-[14px] w-full"
              onClick={handleClickLogoutBtn}
            >
              <LogoutIcon className="mr-3" />
              <p className="text-md font-bold">Logout</p>
            </button>
          </div>
        </div>

        <img src={IMAGES.FOOTER_NAV} alt="Footer Nav" />
      </nav>

      {/* Body Content */}
      {children}
    </div>
  );
};

export default Sidebar;
