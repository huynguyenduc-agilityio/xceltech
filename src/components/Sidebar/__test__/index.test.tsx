import { render } from '@testing-library/react';

import Sidebar from '..';

const setup = () => {
  render(
    <Sidebar>
      <div>Test Content</div>
    </Sidebar>,
  );
};

describe('Sidebar component', () => {
  it('should match snapshot', () => {
    const container = setup();

    expect(container).toMatchSnapshot();
  });
});
