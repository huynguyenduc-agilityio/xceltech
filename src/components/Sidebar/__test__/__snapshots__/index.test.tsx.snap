// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component should match snapshot 1`] = `undefined`;
